src/client/metadataApiDeploy.ts

```diff
@@ -437,20 +436,38 @@ export class MetadataApiDeploy extends MetadataTransfer<
   }
 
   private async getZipBuffer(): Promise<Buffer> {
-    if (this.options.mdapiPath) {
-      if (!fs.existsSync(this.options.mdapiPath) || !fs.lstatSync(this.options.mdapiPath).isDirectory()) {
-        throw messages.createError('error_directory_not_found_or_not_directory', [this.options.mdapiPath]);
+    const mdapiPath = this.options.mdapiPath;
+    if (mdapiPath) {
+      if (!fs.existsSync(mdapiPath) || !fs.lstatSync(mdapiPath).isDirectory()) {
+        throw messages.createError('error_directory_not_found_or_not_directory', [mdapiPath]);
       }
-      // make a zip from the given directory
-      const zip = createArchive('zip', { zlib: { level: 9 } });
-      // anywhere not at the root level is fine
-      zip.directory(this.options.mdapiPath, 'zip');
-      // archiver/zip.finalize looks like it is async, because it extends streams, but it is not meant to be used that way
-      // the typings on it are misleading and unintended.  More info https://github.com/archiverjs/node-archiver/issues/476
-      // If you await it, bad things happen, like the convert process exiting silently.  https://github.com/forcedotcom/cli/issues/1791
-      // leave the void as it is
-      void zip.finalize();
-      return stream2buffer(zip);
+
+      const zip = JSZip();
+
+      const zipDirRecursive = (dir: string): void => {
+        const list = fs.readdirSync(dir);
```

Comment:
`readdir` has an option for [`withFileTypes`](https://nodejs.org/api/fs.html#fspromisesreaddirpath-options) that will give you the isDirectory value without having to do _another_ fs operation inside the loop.